src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpSyntaxNodeOrToken.cs

```diff
@@ -23,19 +23,18 @@ private protected HttpSyntaxNodeOrToken(SourceText sourceText, HttpSyntaxTree? s
 
     public HttpSyntaxNode? Parent { get; internal set; }
 
+    public abstract bool IsSignificant { get; }
+
+    public abstract TextSpan FullSpan { get; }
+
     public abstract TextSpan Span { get; }
 
     public HttpSyntaxTree? SyntaxTree { get; }
 
     /// <summary>
     /// Gets the significant text of the current node or token, without trivia.
     /// </summary>
-    public string Text => TextWithTrivia.Trim();
-
-    /// <summary>
-    /// Gets the text of the current node or token, including trivia.
-    /// </summary>
-    public string TextWithTrivia => SourceText.ToString(Span);
+    public abstract string Text { get; }
```

Comment:
I don't think this needs to be abstract since the implementation can be identical for all derived types.

```suggestion
    public string Text => SourceText.ToString(Span);
    public string FullText => SourceText.ToString(FullSpan);
```

Reply:
Good catch.